fuse/ozone-fuse/main.go

```diff
@@ -0,0 +1,155 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"github.com/apache/ozone-go/api"
+
```

Comment:
This change also adds a fuse client. This should be a separate change with discussion on the FUSE ? We also need a discussion on the consistency provided by the fuse client.

Reply:
Sure thing. I will upload it in a different patch.

I assume the consistency question is about the visibility of the files during the upload. (Let me know if you see any other problems). I agree that we need to investigate the question and define what kind of guarantees can be provided.

By proposal is to start with a read-only fuse driver. It can be useful in many use-cases (like reading (!) ozone keys from the containers) and during the implementation we can understand the challenges about the write part which can be further discussed.